README.md

```diff
@@ -1,2 +1,168 @@
-# ENGAGE-HF-Firebase
+# ENGAGE-HF Firebase
+
 Firebase cloud hosting infrastructure for the ENGAGE-HF project.
+
+# Data Scheme
+
+This document describes how data is stored in Firestore for the Engage-HF app.
+
+## Custom Types
+
+The following section describes custom types defined for this system to be later used
+
+### LocalizedText
+
+A LocalizedText object shall be used whenever text requires localization or may require in the future. The object may either simply be a single string (in the case of localizations not being available yet) or a dictionary with string keys and string values. If LocalizedText is represented by a string-string dictionary, the keys represent language-codes following the ISO 639-1 (or if necessary ISO 639-1) standard and the respective text in the given language as value.
+
+A LocalizedText object cannot be used in FHIR-conforming types due to its incompatibility with the standard. FHIR types commonly contain text in one language only to be specified using the `language` property.
```

Comment:
I assume this is only really referring to `Questionnaires` as we discussed in our last meeting. Or should we rather also include Medications here?

Would be good to add a list of excepted endpoints/types that follow the scheme. And we should define how we add document IDs (probably language prefixes) to easily automatically load localized, e.g., questionnaires here.

Reply:
The provider dashboard will be English for now, but we might need to translate terminologies and additional context (e.g. educational text about the medication) in different languages.

Given the UI mockups, each medication might also have some text and description associated with it and that would be in different languages. That's why I would be leaning toward a localization for medications as well + we might want to use localized strings there to ensure that we have a single medication instance (and ID) but can multiplex for different languages in the relevant medication text fields itself?

I think we might need to add even more "non-standard" FHIR mediation information in here so I am wondering if it rather makes sense to make the medication type a non-FHIR type with a smaller but more manageable structure. Would also be good to get @vishnuravi's input on this.

Reply:
I'd like to discuss this in our data scheme discussion meeting - open for all of the options below (in addition to ones I haven't thought about):

1: We keep the FHIR structure and store all other information as extensions using the LocalizedText object.
2: We keep the FHIR structure, introduce a new table with custom information (e.g. the localizations) and only keep one id to that object in the extension.
3: We create custom objects that have no resemblance of FHIR.

Reply:
Would be great to record the results of the discussion today 👍

Reply:
The discussion preferred the solution to add a Firebase function that would translate the texts using an external service. 

For now, we might also just have that function return the medication as it is stored in the database in English and add the localization part whenever it becomes necessary.